From 188ce6147f882dfbfa88cb0145c5ffc4c5928216 Mon Sep 17 00:00:00 2001
From: Octavian Purdila <octavian.purdila@nxp.com>
Date: Tue, 18 Jul 2017 12:32:14 +0300
Subject: [PATCH 2172/5242] MLK-16020 drivers: soc: imx8: fill in machine
 field

commit  82b7b073463c66c5818d2a376fcbb6fb3927e6ef from
https://source.codeaurora.org/external/imx/linux-imx.git

As for imx6/7, read the 'model' field from device tree and fill in the
machine soc field.

Signed-off-by: Octavian Purdila <octavian.purdila@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/soc/imx/soc-imx8.c |    2 ++
 1 file changed, 2 insertions(+)

diff --git a/drivers/soc/imx/soc-imx8.c b/drivers/soc/imx/soc-imx8.c
index 0e38e4a..e7ff4e5 100644
--- a/drivers/soc/imx/soc-imx8.c
+++ b/drivers/soc/imx/soc-imx8.c
@@ -234,6 +234,8 @@ static int __init imx8_soc_init(void)
 	if (!soc_dev_attr->revision)
 		goto free_soc;
 
+	of_property_read_string(of_root, "model", &soc_dev_attr->machine);
+
 	soc_dev = soc_device_register(soc_dev_attr);
 	if (IS_ERR(soc_dev))
 		goto free_rev;
-- 
1.7.9.5

